src/main/java/nl/knaw/dans/managedeposit/db/DepositPropertiesDAO.java

```diff
@@ -106,44 +110,53 @@ private Predicate buildQueryCriteria(Map<String, List<String>> queryParameters,
-                            orPredicateItem = criteriaBuilder.equal(root.get("depositor"), value);
-                            break;
-
-                        case "deleted":
-                            if (Boolean.parseBoolean(value))
-                                orPredicateItem = criteriaBuilder.isTrue(root.get("deleted"));
-                            else
-                                orPredicateItem = criteriaBuilder.isFalse(root.get("deleted"));
-                            break;
-
-                        case "state":
-                            orPredicateItem = criteriaBuilder.equal(root.get("depositState"), value);
-                            break;
-
-                        case "startdate":
-                        case "enddate":
-                            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
-                            LocalDate date = LocalDate.parse(value, formatter);
-                            var asked_date = OffsetDateTime.of(date.atStartOfDay(), ZoneOffset.UTC);
-
-                            if (parameter.equals("startdate"))
-                                orPredicateItem = criteriaBuilder.greaterThan(root.get("depositCreationTimestamp"), asked_date);
-                            else
-                                orPredicateItem = criteriaBuilder.lessThan(root.get("depositCreationTimestamp"), asked_date);
-                            break;
-
-                        default:
-                            orPredicateItem = criteriaBuilder.equal(root.get(key), value);
-                    }
-                    orPredicatesList.add(orPredicateItem);
+                switch (parameter) {
+                    case "depositid":
+                        orPredicateItem = criteriaBuilder.equal(root.get("depositId"), value);
+                        break;
+
+                    case "user":
+                        orPredicateItem = criteriaBuilder.equal(root.get("depositor"), value);
+                        break;
+
+                    case "deleted":
+                        if (Boolean.parseBoolean(value))
+                            orPredicateItem = criteriaBuilder.isTrue(root.get("deleted"));
+                        else
+                            orPredicateItem = criteriaBuilder.isFalse(root.get("deleted"));
+                        break;
+
+                    case "state":
+                        orPredicateItem = criteriaBuilder.equal(root.get("depositState"), value);
+                        break;
+
+                    case "startdate":
+                    case "enddate":
+                        if (value.isEmpty()) {
+                            orPredicateItem = criteriaBuilder.isNull(root.get("depositCreationTimestamp"));
+                        }
+                        else {
+                            try {
+                                DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
```

Comment:
To keep it simple return code 400 ("invalid or incorrectly formatted parameters ... " as defined by the API) in case of repeated start-date or end-date